### PR TITLE
presto-benchto-benchmarks

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: presto-benchto-benchmarks/pom.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>com.facebook.presto</groupId>
+            <artifactId>presto-jdbc</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>com.teradata.benchto</groupId>
+            <artifactId>benchto-driver</artifactId>
+            <version>0.2-SNAPSHOT</version>
```
Comment (idemura) [inline presto-benchto-benchmarks/pom.xml]: Is depending on snapshot a good practice?

File: presto-benchto-benchmarks/src/main/resources/benchmarks/presto/tpch.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+datasource: presto
+query-names: presto/tpch/${query}.sql
+runs: 6
```
Comment (idemura) [inline presto-benchto-benchmarks/src/main/resources/benchmarks/presto/tpch.yaml]: If we would like to have values for `runs, prewarmup-runs, database` and etc. customized, how we do it?

File: presto-benchto-benchmarks/pom.xml (hunk 1 above)
Comment (electrum) [inline presto-benchto-benchmarks/pom.xml]: It's fine to put it up for review with a snapshot, but we should update to a release version before merging.

(there are often changes needed as the result of the review, so having a snapshot during review is helpful)

File: pom.xml [hunk 3]
```diff
@@ -620,7 +620,7 @@
             <dependency>
                 <groupId>org.apache.commons</groupId>
                 <artifactId>commons-math3</artifactId>
-                <version>3.2</version>
+                <version>3.5</version>
```
Comment (electrum) [inline pom.xml]: Latest version is 3.6.1. Should we use that instead?

Release notes: http://commons.apache.org/proper/commons-math/changes-report.html

File: presto-benchto-benchmarks/src/main/resources/benchmarks/presto/tpch.yaml (hunk 2 above)
Comment (findepi) [inline presto-benchto-benchmarks/src/main/resources/benchmarks/presto/tpch.yaml]: Support for overrides (with a command line switch) would be a good addition to Benchto.

File: pom.xml (hunk 3 above)
Comment (findepi) [inline pom.xml]: I don't insist. I just needed Benchto-driver and Presto agree on the version to get build time check satisfied (and, of course, avoid any problems at run-time). Shall I 3.6.1?

File: presto-benchto-benchmarks/src/main/resources/benchmarks/presto/tpch.yaml (hunk 2 above)
Comment (sopel39) [inline presto-benchto-benchmarks/src/main/resources/benchmarks/presto/tpch.yaml]: https://github.com/prestodb/benchto/issues/37 should solve the issue. For now I think benchmarks could serve as a template

File: presto-benchto-benchmarks/README.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+```
+
+### Running benchto-driver
+
+With the scene set up as in the previous section, the benchmark can be run with:
+```bash
+./mvnw clean package -pl presto-benchto-benchmarks
+java -Xmx1g -jar presto-benchto-benchmarks/target/presto-benchto-benchmarks-*-executable.jar \
+    --activeBenchmarks=presto/tpch --profile=presto-devenv
+```
```
Comment (findepi) [inline presto-benchto-benchmarks/README.md]: @idemura this is the command you can use to run the benchmarks. Thus, if you create `src/main/resources/benchmarks/presto/tpch.facebook.yaml` (as per your question elsewhere), just pass `--activeBenchmarks=presto/tpch.facebook`

(no support for includes/overrides/extensions implemented so far)

File: presto-benchto-benchmarks/src/main/resources/benchmarks/presto/tpch.yaml (hunk 2 above)
Comment (sopel39) [inline presto-benchto-benchmarks/src/main/resources/benchmarks/presto/tpch.yaml]: template for temporary copy of benchmark until we implement: https://github.com/prestodb/benchto/issues/37. 
Another option is to wait for https://github.com/prestodb/benchto/issues/37 and then merge `presto-benchmarks` module. 
Depends how fast you need benchmarks module to land in.

File: presto-benchto-benchmarks/README.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+* An environment [defined in Benchto service](https://github.com/prestodb/benchto/tree/master/benchto-service#creating-environment)
+
+### Configuring benchmarks
+
+Benchto driver needs to know two things: what benchmark is to be run and what environment
+it is to be run on. For the purpose of the following example, we will use `tpch` benchmark
+and Presto server running at `localhost:8080`, with Benchto service running at `localhost:8081`.
+
+Benchto driver uses Sprint Boot to locate environment configuration file, so to pass the
+configuration. To continue with our example, one needs to place an `application-presto-devenv.yml`
```
Comment (idemura) [inline presto-benchto-benchmarks/README.md]: Now I see that it works through spring. Is `application.yaml` (please fix this here! -- you have yml) is read by spring and merged with `application-presto-devenv.yml` by spring? In this case, application.yaml is read, then application-presto-devenv.yaml and merged?

**Comment by findepi** [comment]
Benchto API is going to be changed as a result of prestodb/benchto#37.

@idemura, @sopel39 I'm closing this PR for now. Will be reopened once that issue is developed, aligned with benchto changes.

**Comment by idemura** [comment]
Merged, thanks!

**Pull request closed without merging**